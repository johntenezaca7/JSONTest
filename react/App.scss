$primary-color1: rgb(255, 255, 255);
$primary-font: 'Gugi', cursive;
$second-font :'Roboto', sans-serif;

* {
    box-sizing: border-box
}
body {
    margin: 0%;
    font-family: $second-font;
    background-color: $primary-color1;
    color: rgb(32, 32, 32);
    overflow:hidden;
  }
  img {
    cursor: pointer;
  }
  
  a {
    color: black;
    text-decoration: none;
    cursor: pointer;
  }
  input {
      padding: 0px;
  }

  video {
      position: absolute;
      top: 0;
      left: 0;
      width: 100%;
      overflow: hidden;
      
  }
  .video-overlay {
      height: 100%;
      width: 100%;
      position: absolute;
      top:0;
      left: 0;
      background-color: rgb(53, 87, 126);
      z-index:1;
      opacity: 0.85;   
      overflow: hidden;
  }

  .landing-video{
      display: gird;
      grid-template-columns: auto;
      justify-content: center;
  }

  .landing {
    background-color: inherit;
    position: absolute;
    top:25%;
    left:25%;
    right:25%;
    margin: auto;
    display: grid;
    grid-template-columns: auto;
    z-index: 1;
    color: white;
    align-self: center;

    h2 {
        font-family: $primary-font; 
        font-size: 50px;
        margin: 0%;
        justify-self: center;
    }

    .GetName {
        text-align: center;
        margin-bottom: 2%;
        font-family: $primary-font;
        justify-self: center
    }

    div {
    justify-self: center;
    }
}
  
  .inside-landing input {
    height: 40px;
    font-size: 20px;
    text-align: center;
  }
  .landing-btn {
    margin-top: 5%;
    background-color: rgb(255, 255, 255); 
    border: none;
    width: 100%;
    color: white;
    padding: 0px;
    height: 40px;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    font-size: 16px;
    -webkit-transition-duration: 0.4s; /* Safari */
    transition-duration: 0.4s;
    cursor: pointer;
    justify-self: center
  }
  
  .landing-btn {
    background-color: white;
    color: black;
  }
  
  .landing-btn:hover {
    background-color: rgb(35, 55, 78);
    color: white;
  }
  
  .test-display h2 {
    text-align: center;
    margin-bottom: 5%;
  }
  
  .test-display {
    display: grid;
    grid-template-columns: auto;
    grid-template-rows: 1fr 4fr
  }
  
  .test-box {
    height: 500px;
    width: 900px;
    border: 1px solid rgb(250, 250, 250);
    background-color: white;
    border-radius: 2%;
    box-shadow: -2px 3px 5px rgba(77, 78, 79, 0.2);
    justify-self: center;
  }
  
  .test-details{
    text-align: center;
  }
  .test-details h4 {
    font-size: 25px;
    margin: 1%;
  }
  .test-details p {
    font-size: 16px;
    line-height: 1.5;
  }
  .container {
    padding: 5%;
  }
  .back-n-count {
    display: grid;
    grid-template-columns: 1fr 1fr;
  }
  
  .back-n-count p {
    justify-self: end;
  }
  
  .answers {
    display: flex;
    flex-flow: row wrap;
    justify-content: center
  }
  
  .answer-box {
    border: 1px solid grey;
    width: 180px;
    height: 100px;
    border-radius: 15px;  
    box-shadow: -2px 3px 5px rgba(77, 78, 79, 0.2);
    margin: 1%;
    cursor: pointer;
    display: flex;
    flex-flow: row wrap;
    justify-content: center
  }
  
  .answer-box p {
    align-self: center;
    padding-left: 5px;
  
  }