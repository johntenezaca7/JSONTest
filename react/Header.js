import React from "react";

const Header = Props => {
  return <div>Inital Commit</div>;
};

export default Header;
