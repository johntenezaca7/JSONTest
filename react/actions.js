export const SET_SEARCH_TERM = "SET_SEARCH_TERM";
export const INCREMENT = "INCREMENT";
export const DECREMENT = "DECREMENT";
export const RESTART = "RESTART";
