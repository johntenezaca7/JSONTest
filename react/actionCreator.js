import { SET_SEARCH_TERM, INCREMENT, DECREMENT, RESTART } from "./actions";

export function setSearchTerm(searchTerm) {
  return { type: SET_SEARCH_TERM, payload: searchTerm };
}

export function incrementScore(score) {
  return { type: INCREMENT, payload: score };
}

export function decrementScore(score) {
  return { type: DECREMENT, payload: score };
}

export function restart() {
  return { type: RESTART, payload: 0 };
}
