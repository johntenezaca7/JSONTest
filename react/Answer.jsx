import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import { incrementScore } from "./actionCreator";

const Answer = props => {
  const { choice, incScore, onClick, current } = props;

  const updateScore = () => {
    let score = choice[1];
    incScore(score);
  };

  const fireAll = () => {
    updateScore();
    onClick();
  };
  let isDone;

  if (current.done) {
    isDone = (
      <div className="answer-box" onClick={updateScore}>
        <Link to="/done">
          <p>{choice[0]}</p>
        </Link>
      </div>
    );
  } else {
    isDone = (
      <div className="answer-box" onClick={fireAll}>
        <p>{choice[0]}</p>
      </div>
    );
  }
  return isDone;
};

Answer.propTypes = {
  incScore: PropTypes.func,
  current: PropTypes.object,
  choice: PropTypes.array,
  onClick: PropTypes.func
};

const mapDispatchtoProps = dispatch => ({
  incScore(score) {
    dispatch(incrementScore(score));
  }
});

export const Unwrapped = Answer;

export default connect(null, mapDispatchtoProps)(Answer);
