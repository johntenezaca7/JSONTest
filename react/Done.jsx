import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import { restart } from "./actionCreator";

const Done = props => {
  const { score, takeTestAgain } = props;
  return (
    <div className="done">
      <h1>Your score is {score} out of 20.</h1>
      <button onClick={takeTestAgain}>
        <Link to="/assessment">
          Take it again!
        </Link>
      </button>
    </div>
  );
};

const mapStateToProps = state => ({ score: state.score });

const mapDispatchtoProps = dispatch => ({
  takeTestAgain() {
    dispatch(restart());
  }
});

Done.propTypes = {
  score: PropTypes.number,
  takeTestAgain: PropTypes.func
};

export default connect(mapStateToProps, mapDispatchtoProps)(Done);
