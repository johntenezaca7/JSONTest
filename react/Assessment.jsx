import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { decrementScore } from "./actionCreator";
import Answer from "./Answer";

class Assessment extends Component {
  constructor(props) {
    super(props);
    this.state = {
      index: 0,
      testNum: 1,
      current: this.props.questions[0],
      savedScore: ""
    };

    this._onRightClick = this._onRightClick.bind(this);
    this._onLeftClick = this._onLeftClick.bind(this);
    this._run = this._run.bind(this);
  }

  _run(direction) {
    let nextValue, questions;

    if (direction === "right") {
      if (this.state.index <= 4) {
        this.state.index += 1;
        this.state.testNum += 1;

        nextValue = this.state.index;
        questions = this.props.questions;

        this.setState((prev, next) => {
          let savedVal = prev.savedScore ? prev.savedScore : next.currentScore;
          return {
            current: questions[nextValue],
            savedScore: savedVal
          };
        });
      }
    } else if (direction === "left") {
      if (this.state.index !== 0) {
        this.state.index -= 1;
        this.state.testNum -= 1;

        nextValue = this.state.index;
        questions = this.props.questions;

        this.setState((prev, next) => {
          return {
            current: questions[nextValue],
            savedScore: prev.savedScore
          };
        });
      }
    }
  }

  _onRightClick() {
    this._run("right");
  }

  _onLeftClick() {
    this._run("left");
    this.props.decScore(this.state.savedScore);
  }

  render() {
    const { _onLeftClick, _onRightClick } = this;
    const { testNum, current } = this.state;
    const { name } = this.props;

    let checkName;
    let checkNav;

    if (name) {
      checkName = name;
    } else {
      checkName = "Anonymous";
    }

    if (testNum === 1) {
      checkNav = <div />;
    } else {
      checkNav = (
        <img
          onClick={_onLeftClick}
          src="https://image.ibb.co/jyA5n7/prev.png"
          alt="Back"
        />
      );
    }
    return (
      <div className="test-display">
        <h2>Welcome {checkName}!</h2>
        <div className="test-box">
          <div className="container">
            <div className="back-n-count">
              {checkNav}
              <p>Step {testNum} of 4</p>
            </div>
            <div className="test-details">
              <h4>{current.title}</h4>
              <p>{current.description}</p>
            </div>
            <div className="answers">
              {current.choices.map((ele, i) => {
                return (
                  <Answer
                    choice={ele}
                    key={i}
                    onClick={_onRightClick}
                    current={current}
                  />
                );
              })}
            </div>
          </div>
        </div>
      </div>
    );
  }
}

Assessment.propTypes = {
  currentScore: PropTypes.number,
  decScore: PropTypes.func,
  questions: PropTypes.array,
  name: PropTypes.string
};

const mapStateToProps = state => ({
  questions: state.questions,
  name: state.searchTerm,
  currentScore: state.score
});

const mapDispatchtoProps = dispatch => ({
  decScore(score) {
    dispatch(decrementScore(score));
  }
});

export const Unwrapped = Assessment;

export default connect(mapStateToProps, mapDispatchtoProps)(Assessment);
