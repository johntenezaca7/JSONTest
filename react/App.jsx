import React from "react";
import { Route } from "react-router-dom";
import { Provider } from "react-redux";
import store from "./store";

import Landing from "./Landing";
import Test from "./Assessment";
import Done from "./Done";

const App = () => (
  <Provider store={store}>
    <div className="app">
      <Route exact path="/" component={props => <Landing />} />
      <Route exact path="/assessment" component={props => <Test />} />
      <Route exact path="/done" component={props => <Done />} />
    </div>
  </Provider>
);

export default App;
