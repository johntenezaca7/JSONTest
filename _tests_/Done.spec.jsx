import React from "react";
import jest from "jest";
import { shallow } from "enzyme";
import { Unwrapped as UnwrappedDone } from "../react/Done";
import { StaticRouter } from "react-router-dom";

test("Done Component renders correctly", () => {
  const component = shallow(
    <UnwrappedDone score={20} takeTestAgain={f => f} />
  );
  expect(component).toMatchSnapshot();
});
