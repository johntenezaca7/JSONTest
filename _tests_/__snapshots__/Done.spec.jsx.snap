// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Done Component renders correctly 1`] = `
<div
  className="done"
>
  <h1>
    Your score is 
    20
     out of 20.
  </h1>
  <button
    onClick={[Function]}
  >
    <Link
      replace={false}
      to="/assessment"
    >
      Take it again!
    </Link>
  </button>
</div>
`;
